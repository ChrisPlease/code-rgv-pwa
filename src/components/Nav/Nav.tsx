import * as React from "react";

/**
 * Navigation Component
 */
export class Nav extends React.Component {
  public render() {
    return <div>Nav</div>;
  }
}
