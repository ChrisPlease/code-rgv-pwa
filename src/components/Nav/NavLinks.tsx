/**
 * Navigation Links
 */
import * as React from "react";
import { Link } from "react-router-dom";

export class NavLinks extends React.PureComponent {
  public render() {
    return (
      <div className="navbar-start">
        <Link to="/" className="navbar-item">
          Home
        </Link>
        <Link to="/events" className="navbar-item">
          Events
        </Link>
        <Link to="/projects" className="navbar-item">
          Projects
        </Link>
        <div className="navbar-item has-dropdown is-hoverable">
          <Link to="/about/about-us" className="navbar-link is-active">
            About
          </Link>
          <div className="navbar-dropdown">
            <Link
              to="/about/about-us"
              className="navbar-item"
              role="button"
              rel="nooopener noreferrer"
            >
              About Us
            </Link>
            <a
              href="https://github.com/CodeRGV/codeofconduct"
              className="navbar-item"
              role="button"
              rel="nooopener noreferrer"
            >
              Code Of Conduct
            </a>
            <Link
              to="/about/charter"
              className="navbar-item"
              role="button"
              rel="nooopener noreferrer"
            >
              Charter
            </Link>
            <Link
              to="/about/strategic-plan"
              className="navbar-item"
              role="button"
              rel="nooopener noreferrer"
            >
              Strategic Plan
            </Link>
          </div>
        </div>
      </div>
    );
  }
}
