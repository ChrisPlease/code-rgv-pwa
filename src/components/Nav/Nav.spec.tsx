/**
 * Navbar test
 */
import { shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";
import * as React from "react";
import { Nav } from "./Nav";

test("Navbar exists", () => {
  const isBurgerActive = true;
  const dispatch = () => console.info("test");
  const component = renderer.create(
    <MemoryRouter>
      <Nav isBurgerActive dispatch={dispatch} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
