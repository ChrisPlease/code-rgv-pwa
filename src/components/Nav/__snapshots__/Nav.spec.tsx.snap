// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar exists 1`] = `
<nav
  className="navbar"
  style={
    Object {
      "border": "1px solid #ffaa00",
    }
  }
>
  <div
    className="navbar-brand"
  >
    <a
      className="navbar-item"
      href={undefined}
      rel="noopener noreferrer"
      role="button"
      target="_blank"
    >
      <img
        alt="brownsville logo"
        src="test-file-stub"
      />
      <span
        className="title is-4"
      >
        <strong>
          CodeRGV
        </strong>
      </span>
    </a>
    <div
      className="navbar-burger "
      data-target="codeRgvMenu"
      onClick={[Function]}
      role="button"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="navbar-menu is-active"
    id="codeRgvMenu"
  >
    <div
      className="navbar-start"
    >
      <a
        className="navbar-item"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="navbar-item"
        href="/events"
        onClick={[Function]}
      >
        Events
      </a>
      <a
        className="navbar-item"
        href="/projects"
        onClick={[Function]}
      >
        Projects
      </a>
      <div
        className="navbar-item has-dropdown is-hoverable"
      >
        <a
          className="navbar-link is-active"
          href="/about/about-us"
          onClick={[Function]}
        >
          About
        </a>
        <div
          className="navbar-dropdown"
        >
          <a
            className="navbar-item"
            href="/about/about-us"
            onClick={[Function]}
            rel="nooopener noreferrer"
            role="button"
          >
            About Us
          </a>
          <a
            className="navbar-item"
            href="/about/code-of-conduct"
            onClick={[Function]}
            rel="nooopener noreferrer"
            role="button"
          >
            Code Of Conduct
          </a>
          <a
            className="navbar-item"
            href="/about/charter"
            onClick={[Function]}
            rel="nooopener noreferrer"
            role="button"
          >
            Charter
          </a>
          <a
            className="navbar-item"
            href="/about/strategic-plan"
            onClick={[Function]}
            rel="nooopener noreferrer"
            role="button"
          >
            Strategic Plan
          </a>
        </div>
      </div>
    </div>
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item"
        href="https://github.com/codergvbrownsville"
        rel="noopener noreferrer"
        role="button"
        target="_blank"
      >
        <i
          aria-hidden="true"
          className="fa fa-github"
        />
      </a>
      <a
        className="navbar-item"
        href="https://www.facebook.com/codergvbrownsville/"
        rel="noopener noreferrer"
        role="button"
        target="_blank"
      >
        <i
          aria-hidden="true"
          className="fa fa-facebook-square"
        />
      </a>
      <a
        className="navbar-item"
        href="https://codergv.herokuapp.com/"
        rel="noopener noreferrer"
        role="button"
        target="_blank"
      >
        <i
          aria-hidden="true"
          className="fa fa-slack"
        />
      </a>
    </div>
  </div>
</nav>
`;
